Dev

AntonKut22 opened the pull request:
Поправлена логика, и фронт.
Тесты и логирование пока не прикручены

AntonKut22 committed:
re-Upload Project

AntonKut22 committed:
refactor logic

AntonKut22 committed:
delete InitGameFamily.java

AntonKut22 committed:
add Bootstrap

AntonKut22 committed:
added pictures

AntonKut22 committed:
added pictures

AntonKut22 committed:
added logs

Anderwerty commented on pom.xml:
```diff
@@ -4,26 +4,32 @@
     <modelVersion>4.0.0</modelVersion>
 
-    <groupId>ru</groupId>
+    <groupId>com.example</groupId>
```
It should be `ru.javarush`

AntonKut22 replied:
Fixed

Anderwerty commented on pom.xml:
```diff
@@ -4,26 +4,32 @@
-            <version>5.0.0</version>
+            <groupId>javax.servlet</groupId>
+            <artifactId>javax.servlet-api</artifactId>
+            <version>4.0.1</version>
```
please extract the version of all dependencies into maven properties

AntonKut22 replied:
Fixed

Anderwerty commented on src/main/java/ru/javarush/family/controller/DispatcherServlet.java:
```diff
@@ -0,0 +1,119 @@
+
+    @Override
+    public void init() throws ServletException {
+        //TODO починить, чтобы работало через относительный путь
```
Please even TODO comments write using English

AntonKut22 replied:
Fixed

Anderwerty commented on src/main/java/ru/javarush/family/entitie/User.java:
```diff
@@ -0,0 +1,27 @@
+package ru.javarush.family.entitie;
```
please fix typo for package `entitie` ----> `entity`

AntonKut22 replied:
Fixed

Anderwerty commented on src/main/java/ru/javarush/family/entitie/User.java:
```diff
@@ -0,0 +1,27 @@
+public class User {
+
+    @JsonProperty("role")
+    Role role;
```
please make field private

AntonKut22 replied:
Fixed

Anderwerty commented on src/main/java/ru/javarush/family/controller/DispatcherServlet.java:
```diff
@@ -0,0 +1,119 @@
+    public void init() throws ServletException {
+        //TODO починить, чтобы работало через относительный путь
+        log.info("Start DispatcherServlet {}");
+        File fileUser = new File("D:\\Обучение\\JRU\\ru.familyQuest\\family\\src\\main\\resources\\users.json");
```
We need to use relative file path

AntonKut22 replied:
Fixed

Anderwerty commented on src/main/java/ru/javarush/family/controller/DispatcherServlet.java:
```diff
@@ -0,0 +1,119 @@
+        //TODO починить, чтобы работало через относительный путь
+        log.info("Start DispatcherServlet {}");
+        File fileUser = new File("D:\\Обучение\\JRU\\ru.familyQuest\\family\\src\\main\\resources\\users.json");
+        File fileQuestion = new File("D:\\Обучение\\JRU\\ru.familyQuest\\family\\src\\main\\resources\\questions.json");
```
the same here

AntonKut22 replied:
Fixed

Anderwerty commented on src/main/java/ru/javarush/family/controller/DispatcherServlet.java:
```diff
@@ -0,0 +1,119 @@
+                ", questions=" + questions.getCountQuestion() +
+                '}';
+    }
+}
```
After last brace should be a blank line

Anderwerty commented on src/main/java/ru/javarush/family/controller/DispatcherServlet.java:
```diff
@@ -0,0 +1,119 @@
+package ru.javarush.family.controller;
+
+import lombok.ToString;
```
Do we need this annotation?

Anderwerty commented on src/main/java/ru/javarush/family/controller/DispatcherServlet.java:
```diff
@@ -0,0 +1,119 @@
+    }
+
+    @Override
+    public String toString() {
```
Why do we need this method here?

Anderwerty commented on src/main/java/ru/javarush/family/repository/Questions.java:
```diff
@@ -0,0 +1,54 @@
+
+@Getter
+public class Questions {
+    private static final Logger log = LogManager.getLogger(Questions.class);
```
please add a blank line between instance of log and field for readability

AntonKut22 replied:
Fixed

Anderwerty commented on src/main/java/ru/javarush/family/repository/Questions.java:
```diff
@@ -0,0 +1,54 @@
+    }
+
+    private Map<Long, Question> initialisationUsers(File file) {
+        ObjectMapper objectMapper = new ObjectMapper();
```
Let's inject objectMapper as a field. We don't need to create `ObjectMapper` for each method invocation

AntonKut22 replied:
Fixed

Anderwerty commented on src/main/java/ru/javarush/family/repository/Questions.java:
```diff
@@ -0,0 +1,54 @@
+    }
+
+    @Override
+    public String toString() {
```
do we need it?

AntonKut22 replied:
I agree, not very necessary information. Deleted

Anderwerty commented on src/main/java/ru/javarush/family/repository/Questions.java:
```diff
@@ -0,0 +1,54 @@
+@Getter
+public class Questions {
+    private static final Logger log = LogManager.getLogger(Questions.class);
+    private Map<Long, Question> questionsMap;
```
using the word `map` in the map naming looks like Hungarian notation
let's use pattern `keyToValue` or `keyWithValue` or `keyByValue`
We can name it as idToQuestion

AntonKut22 replied:
renamed to "idToQuestion"

Anderwerty commented on src/main/java/ru/javarush/family/repository/Users.java:
```diff
@@ -0,0 +1,70 @@
+import java.util.Map;
+
+@Getter
+public class Users {
```
the same about current class

AntonKut22 replied:
Fixed

Anderwerty reviewed (commented):
Please fix issues in code

AntonKut22 committed:
insignificant refactor

AntonKut22 committed:
corrections after checking Andrey

AntonKut22 committed:
added part of the tests

AntonKut22 committed:
The project is ready for final verification

AntonKut22 committed:
Update README.md

AntonKut22 committed:
Add Jacoco.
bringing the test coverage to 100%.

AntonKut22 committed:
Merge remote-tracking branch 'origin/dev' into dev

# Conflicts:
#	README.md

AntonKut22 committed:
Fixed Bag.
(After renaming in entity.Ansver "answer" to "textAnswer", the path in has not been fixed in game.jsp
